disable a feature toggle that is on by default

mjseaman opened the pull request:
We document how to enable a feature toggle, but not how to disable one. This adds the config setting to disable a toggle that is on by default.

Would appreciate a check on whether the syntax is correct, and also confirm that this is config functionality we want users to use.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

mjseaman committed:
disable a feature toggle that is on by default

We document how to enable a feature toggle, but not how to disable one. This adds the config setting to disable a toggle that is on by default.

Elfo404 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -2220,6 +2220,10 @@ For more information about Grafana Enterprise, refer to [Grafana Enterprise]({{<
 
 Keys of alpha features to enable, separated by space.
```
given alpha features are not the only one that can be enabled:
```suggestion
Keys of features to enable, separated by space.
```

mjseaman committed:
remove "alpha"

Alpha features are not the only ones that can be enabled, and we don't use the term "alpha" anymore.

Co-authored-by: Giordano Ricci <me@giordanoricci.com>

Elfo404 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -2220,6 +2220,10 @@ For more information about Grafana Enterprise, refer to [Grafana Enterprise]({{<
 
+### FEATURE_TOGGLE_NAME = false 
+  
+Some feature toggles for stable features are on by default. Use this setting to explicitly disable an on-by-default feature toggle with the name FEATURE_TOGGLE_NAME, for example `exploreMixedDatasource = false`.
```
we may also make this a but more explicit, like this maybe (but its pretty inconsistent compared to the rest of the page)

![image](https://github.com/grafana/grafana/assets/1170767/826e6861-b8a9-4e8c-8b56-00e479e4adac)

ryantxu replied:
there is similar language in:
https://github.com/grafana/grafana/blob/main/conf/defaults.ini#L1492

that gets included in the depolyments

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -2218,7 +2218,11 @@ For more information about Grafana Enterprise, refer to [Grafana Enterprise]({{<
+
+### FEATURE_TOGGLE_NAME = false 
+  
+Some feature toggles for stable features are on by default. Use this setting to explicitly disable an on-by-default feature toggle with the name FEATURE_TOGGLE_NAME, for example `exploreMixedDatasource = false`.
```
```suggestion
Some feature toggles for stable features are on by default. Use this setting to disable an on-by-default feature toggle with the name FEATURE_TOGGLE_NAME, for example, `exploreMixedDatasource = false`.
```

chri2547 reviewed (approved):
A small suggestion. Thanks for the contribution!

mjseaman committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

chri2547 committed:
Merge remote-tracking branch 'origin/main' into docs/disable-feature-toggle

chri2547 committed:
makes prettier
